buildSrc/src/main/java/org/elasticsearch/gradle/FileSystemOperationsAware.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.gradle.api.tasks.WorkResult;
+
+/**
+ * A marker interface for tasks signaling that they support basic file operations.
+ * Methods will be added as needed.
+ */
+
+public interface FileSystemOperationsAware {
+    WorkResult delete(Object... objects);
```

Comment:
Not really a "marker" interface if it defines methods. I'm not sure this provides a lot of benefit, as is. Perhaps we can have the interface define the injected method to be implemented by supporting types then provide default method impls that delegate to `getFileSystemOperations()`?

Reply:
My idea when introducing this was more to keep a common pattern for those methods defined by this interface. The injected methods are usually protected and I think we should keep it like that. interfaces do not support protected methods. Also e.g. in RestIntegTest we use the `protected getFileSystemOperations()` coming from `Test` so we cannot change its visibility.